### PR TITLE
docs: Add release process documentation

### PR DESCRIPTION
**Pull request by fabiand**
A small doc describing the formal release process.

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

### CODE REVIEW

**Comment by fabiand** [comment]
@ifireball good point, added a section on the content. Please review.

File: docs/release.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+===============
+
+The goal of this document is to define the release process of KubeVirt, but it
+is not intended to define any release criteria.
+
+
+Overview
+--------
+- KubeVirt uses [semantic versioning](http://semver.org)
+- Primary artefact is the source tree in form of a signed tap using
```
Comment (stu-gott) [inline docs/release.md]: typo: artifact

File: docs/release.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+-------
+The primary artifact of a release is the source tree itself. The trust on the
+tree is established by using _git-evtag_ which can be used to sign the tree
+recursively, including blobs and submodules.
+
+For convenience a number of binary artifacts can be provided alongside a
+release, those are:
+
+- Container images (currently docker images)
+- Client sided binaries (i.e. _virtctl_)
```
Comment (stu-gott) [inline docs/release.md]: typo: "side"

File: docs/release.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+
+Cadence
+-------
+A release is taking place on every first Monday of a month.
```
Comment (stu-gott) [inline docs/release.md]: Your intended meaning is clear, but I suggest rewording to future tense: "Releases will take place on the first Monday of each month."

File: docs/release.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+
+Releasing
+---------
+The release process is mostly automatic and consists of the following steps:
+
+1. Tag a commit using `git evtag sign $TAG` (which is a signed and annotated
+   tag)
+   1. Provide the release notes as part of the tag commit message
+2. Push the tag to github `git push origin $TAG`
```
Comment (stu-gott) [inline docs/release.md]: It's probably obvious, but it might be worth noting that `origin` in this case refers to `https://github.com/kubevirt/kubevirt.git`

File: docs/release.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+- Container images (currently docker images)
+- Client sided binaries (i.e. _virtctl_)
+
+
+Cadence
+-------
+A release is taking place on every first Monday of a month.
+The release owner is in charge of delaying a release if it is _really_
+necessary.
+If a release has to be delayed by more than a week, then the release must be
```
Comment (rmohr) [inline docs/release.md]: Would be "for more" better?

File: docs/release.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+Content
+-------
+The primary artifact of a release is the source tree itself. The trust on the
+tree is established by using _git-evtag_ which can be used to sign the tree
+recursively, including blobs and submodules.
+
+For convenience a number of binary artifacts can be provided alongside a
+release, those are:
+
+- Container images (currently docker images)
```
Comment (rmohr) [inline docs/release.md]: Shouldn't the images be published on a registry and tagged? Adding one big manifest file (for the beginning non-ha, not so secure, ...) might be a better choice. Here is an example of a weave release with yamls included. Maybe that makes sense for us.

File: docs/release.md (hunk 1 above)
Comment (fabiand) [reply docs/release.md]: Done

File: docs/release.md (hunk 2 above)
Comment (fabiand) [reply docs/release.md]: Done

File: docs/release.md (hunk 3 above)
Comment (fabiand) [reply docs/release.md]: Thanks! Done

File: docs/release.md (hunk 4 above)
Comment (fabiand) [reply docs/release.md]: Yes, good point, I'll try to find the right words here.

File: docs/release.md (hunk 6 above)
Comment (fabiand) [reply docs/release.md]: This sentence should not imply that we publish the containers as binaries/tarballs. I rather intend to say that the release is the source code, but we provide aartifacts - like containers and bianries - in the relevant channels. Thus containers on dockerhub and binaries on the github release page.
I added a sentence to make this clearer.

WRT manifests - For now they are part of the repo.
And yes, we want to also publish them in a dedicated and compiled form for releases - I'm also working on something on that side.

File: docs/release.md (hunk 5 above)
Comment (fabiand) [reply docs/release.md]: Just a matter of taste according to a native speaker :)
